index.js

```diff
@@ -1,85 +1,301 @@
-import { NativeEventEmitter, NativeModules } from 'react-native';
+import { NativeEventEmitter, NativeModules } from "react-native";
+import AppsFlyerConstants from "./PurchaseConnector/constants/constants";
+import InAppPurchaseValidationResult from "./PurchaseConnector/models/in_app_purchase_validation_result";
+import ValidationFailureData from "./PurchaseConnector/models/validation_failure_data";
+import SubscriptionValidationResult from "./PurchaseConnector/models/subscription_validation_result";
 
 const { RNAppsFlyer } = NativeModules;
 const appsFlyer = {};
 const eventsMap = {};
 const appsFlyerEventEmitter = new NativeEventEmitter(RNAppsFlyer);
 
+//Purchase Connector native bridge objects
+const { PCAppsFlyer } = NativeModules;
+const AppsFlyerPurchaseConnector = {};
+const pcEventsMap = {};
+const purchaseConnectorEventEmitter = new NativeEventEmitter(PCAppsFlyer);
+
+function startObservingTransactions() {
+  PCAppsFlyer.startObservingTransactions();
+}
+
+AppsFlyerPurchaseConnector.startObservingTransactions =
+  startObservingTransactions;
+
+function stopObservingTransactions() {
+  PCAppsFlyer.stopObservingTransactions();
+}
+
+AppsFlyerPurchaseConnector.stopObservingTransactions =
+  stopObservingTransactions;
+
+AppsFlyerPurchaseConnector.onSubscriptionValidationResultSuccess = (
+  onSuccess
+) => {
+  if (typeof onSuccess !== "function") {
+    throw new Error("onSuccess callback must be a function");
+  }
+
+  const subValidationSuccessListener =
+    purchaseConnectorEventEmitter.addListener(
+      AppsFlyerConstants.SUBSCRIPTION_VALIDATION_SUCCESS,
+      (result) => {
+        try {
+          const firstEntryValue = Object.values(result)[0];
+          const validationResult =
+            SubscriptionValidationResult.fromJson(firstEntryValue);
+          onSuccess(validationResult);
+        } catch (error) {
+          console.error(
+            "Failed to parse subscription validation result:",
+            error
+          );
+        }
+      }
+    );
```

Comment:
The value from native is a map `[purchaseToken: ValidationResult]`, so you should go over each entry and return a map with the same keys mapped to the parsed JS object representing `ValidationResult`